File: src/cycod/CycoDevProgramRunner.cs
using System;
using System.Reflection.Metadata.Ecma335;
using System.Threading.Tasks;
using ProcessExecution;

public class CycoDevProgramRunner : ProgramRunner
{
    public static async Task<int> RunAsync(string[] args)
    {
        try
        {
            var program = new CycoDevProgramRunner();
            return await program.RunProgramAsync(args);
        }
        finally
        {
            NamedShellProcessManager.Instance.ShutdownAllShells();
        }
    }

    override protected bool ParseCommandLine(string[] args, out CommandLineOptions? commandLineOptions, out CommandLineException? ex)
    {
        return CycoDevCommandLineOptions.Parse(args, out commandLineOptions, out ex);
    }

    private CycoDevProgramInfo _programInfo = new();
}

File: src/cycodjs/src/CycoDevProgramRunner.ts
import { ProgramRunner, CommandLineOptions, CommandLineException, ShellSession } from 'cycod-common';
import { CycoDevCommandLineOptions } from './CommandLine/CycoDevCommandLineOptions';
import { CycoDevProgramInfo } from './CycoDevProgramInfo';

export class CycoDevProgramRunner extends ProgramRunner {
    private _programInfo = new CycoDevProgramInfo();

    static async runAsync(args: string[]): Promise<number> {
        try {
            const program = new CycoDevProgramRunner();
            return await program.runProgramAsync(args);
        } finally {
            ShellSession.disposeAll();
        }
    }

    protected parseCommandLine(args: string[]): { 
        success: boolean; 
        commandLineOptions?: CommandLineOptions; 
        exception?: CommandLineException;
    } {
        return CycoDevCommandLineOptions.parse(args);
    }

    protected displayBanner(): void {
        // TODO: Implement cycod banner display
        console.log('CycoDJS - AI-powered CLI');
    }
}